util/doc.ts

```diff
@@ -0,0 +1,94 @@
+import type { ModuleIndexWithDoc } from "$doc_components/module_index.tsx";
+import type { DocNode } from "https://deno.land/x/deno_doc@v0.34.0/lib/types.d.ts";
+import { getIndex } from "$doc_components/doc.ts";
+import { fileTypeFromURL } from "./registry_utils.ts";
+export type { DocNode };
+
+export interface Index {
+  index: ModuleIndexWithDoc;
+  indexModule: string | undefined;
+  nodes: DocNode[];
+}
+
+export async function getDocs(
+  name: string,
+  version: string,
+  path: string,
+): Promise<DocNode[] | Index> {
+  const type = fileTypeFromURL(path);
+  if (
+    type === "javascript" || type === "typescript" ||
+    type === "tsx" || type === "jsx"
+  ) {
+    return getDocNodes(name, version, path);
+  } else { // TODO: check if it is a directory
```

Comment:
Should probably be addressed before landing this PR?

Reply:
its fine if not, as the endpoint will just error, but ideally, yea. unsure how exactly this could be done though